### ISSUE TITLE
make EB installation as zipped eggs work

### ISSUE THREAD

**Issue by boegel** [hpcugent/easybuild-framework#1369]
The major advantage of this is significant speedup when EasyBuild is installed on a shared filesystem.

The major problem with this is that there's no direct access to the Python modules or easyconfig files anymore as files.
Via `pkg_resources.resource_string` for example, the contents of a specific Python module or data file (like easyconfig files) can be retrieved from a zipped egg, see https://pythonhosted.org/setuptools/pkg_resources.html#resourcemanager-api .

It also introduces a dependency on `setuptools`, since `pkg_resources` is a part of `setuptools`.

However, since the performance impact of this is potentially huge (especially for things like `--search`), this may be worth the trouble...

**Comment by ehiggs** [comment]
:+1: 

I don't think the easyconfig files need to be in the zip just for search to be quicker. This is a problem even for program startup since python really hammers the file system trying to figure out where all the files are. e.g. I just used eb to install a python package and it took 10 minutes(!) Here's output from `strace -fc ...`:

```
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 82.50   35.392959       46145       767       278 wait4
 13.31    5.711826          44    131181     84202 stat
  2.75    1.179423          17     68219     46247 open
  0.72    0.308792         605       510           chmod
  0.35    0.148895           8     17522       231 lstat
  0.28    0.120349         227       530        11 unlink
  0.05    0.023009           1     43214     18545 close
  0.03    0.012201           2      7240           getdents
  0.01    0.002483           0     24122        15 read
  0.00    0.001122           1      1851           poll
```

**Comment by boegel** [comment]
https://github.com/hpcugent/easybuild-framework/pull/1357 will improve that significantly...

Can you check again with using `python -O` in `eb`, see how much of a differenc it makes?

**Comment by boegel** [comment]
Relevant benchmarking for `vsc-base`: https://gist.github.com/boegel/e56c7fa0961a3211a4a0

**Comment by boegel** [comment]
will be fixed in #1399
